code/modules/mob/living/blood.dm

```diff
@@ -2,20 +2,229 @@
+		return
+	time_applied = 0
+	// Non-humans stop bleeding a lot quicker, even if it is not a minor cut
+	if (!ishuman(owner))
+		bleed_rate -= BLEED_HEAL_RATE_MINOR * 4
+	// Make sure to update our icon
+	update_icon()
+	// Set the rate at which we process, so we bleed more on the ground when heavy bleeding
+	tick_interval = bleed_rate <= BLEED_RATE_MINOR ? 1 SECONDS : 0.2 SECONDS
+	// Reduce the actual rate of bleeding
+	if (ishuman(owner))
+		if (bleed_rate > 0 && bleed_rate < BLEED_RATE_MINOR)
+			bleed_rate -= BLEED_HEAL_RATE_MINOR
+		else
+			bandaged_bleeding -= BLEED_HEAL_RATE_MINOR
+	// We have finished bleeding
+	if (bleed_rate <= 0 && bandaged_bleeding <= 0)
+		qdel(src)
+		return
+	// The actual rate of bleeding, can be reduced by holding wounds
+	var/final_bleed_rate = bleed_rate
+	if (HAS_TRAIT(owner, TRAIT_BLEED_HELD))
+		final_bleed_rate = max(0, final_bleed_rate - BLEED_RATE_MINOR)
+	// We aren't actually bleeding
+	if (final_bleed_rate <= 0)
 		return
+	// Actually do the bleeding
+	owner.bleed(min(MAX_BLEED_RATE, final_bleed_rate))
+
+/datum/status_effect/bleeding/proc/update_icon()
+	// The actual rate of bleeding, can be reduced by holding wounds
+	var/final_bleed_rate = bleed_rate
+	if (HAS_TRAIT(owner, TRAIT_BLEED_HELD))
+		final_bleed_rate = max(0, final_bleed_rate - BLEED_RATE_MINOR)
+	// Calculate the message to show to the user
+	if (HAS_TRAIT(owner, TRAIT_BLEED_HELD))
+		linked_alert.name = "Bleeding (Held)"
+		if (bleed_rate > BLEED_RATE_MINOR)
+			linked_alert.desc = "You have serious wounds which are unlikely to heal themselves. You are applying pressure to them, slowing the rate of blood loss."
+		else
+			linked_alert.desc = "You are bleeding and are applying pressure to the wounds, preventing blood from pouring out."
+		linked_alert.icon_state = "bleed_held"
+	else if (bleed_rate == 0 && bandaged_bleeding > 0)
+		linked_alert.name = "Bleeding (Bandaged)"
+		linked_alert.desc = "You have bandages covering your wounds. They will heal slowly if they are not cauterized."
+		linked_alert.icon_state = "bleed_bandage"
 	else
-		bleedsuppress = TRUE
-		addtimer(CALLBACK(src, PROC_REF(resume_bleeding)), amount)
+		if (bleed_rate < BLEED_RATE_MINOR)
+			linked_alert.name = "Bleeding (Light)"
+			linked_alert.desc = "You have some minor cuts that look like they will heal themselves if you don't run out of blood first.[ishuman(owner) ? " Click to apply pressure to the wounds." : ""]"
+			linked_alert.icon_state = "bleed"
+		else
+			linked_alert.name = "Bleeding (Heavy)"
+			linked_alert.desc = "Your wounds are bleeding heavily and are unlikely to heal themselves. Seek medical attention immediately![ishuman(owner) ? " Click to apply pressure to the wounds." : ""]"
+			linked_alert.icon_state = "bleed_heavy"
+	// Set the text to the final bleed rate
+	var/rate_string = "[round(final_bleed_rate, 0.1)]"
+	if (length(rate_string) == 1)
+		rate_string = "[rate_string].0"
+	linked_alert.maptext = MAPTEXT("[rate_string]/s")
```

Comment:
The suggested code doesn't work since it adds a decimal to 11.3 but not to 4, although the original code didn't work for values over 10 anyway. The correct solution is to apply modulus.

I prefer the original code over the suggested edit since it does less work overall, but I'll apply the fix for the decimal. There might be a proc that does that anyway